endpoints/openrtb2/amp_auction.go

```diff
@@ -0,0 +1,236 @@
+package openrtb2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/julienschmidt/httprouter"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/exchange"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbs"
+	"github.com/prebid/prebid-server/pbsmetrics"
+	"github.com/prebid/prebid-server/stored_requests"
+	"net/http"
+	"time"
+)
+
+type AmpResponse struct {
+	Targeting map[string]string `json:"targeting"`
+}
+
+// We need to modify the OpenRTB endpoint to handle AMP requests. This will basically modify the parsing
+// of the request, and the return value, using the OpenRTB machinery to handle everything inbetween.
+func NewAmpEndpoint(ex exchange.Exchange, validator openrtb_ext.BidderParamValidator, requestsById stored_requests.Fetcher, cfg *config.Configuration, met *pbsmetrics.Metrics) (httprouter.Handle, error) {
+	if ex == nil || validator == nil || requestsById == nil || cfg == nil || met == nil {
+		return nil, errors.New("NewEndpoint requires non-nil arguments.")
+	}
+
+	return httprouter.Handle((&endpointDeps{ex, validator, requestsById, cfg, met}).AmpAuction), nil
+}
+
+func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	// Prebid Server interprets request.tmax to be the maximum amount of time that a caller is willing
+	// to wait for bids. However, tmax may be defined in the Stored Request data.
+	//
+	// If so, then the trip to the backend might use a significant amount of this time.
+	// We can respect timeouts more accurately if we note the *real* start time, and use it
+	// to compute the auction timeout.
+
+	deps.metrics.RequestMeter.Mark(1)
```

Comment:
Let's drop this. This meter was only meant to count `/openrtb2/auction` endpoint requests.

Reply:
That is the legacy meter. We have an explicit OpenRTB meter as well. So we should also drop the legacy counter from OpenRTB, but that is outside the scope of this PR.